core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -76,8 +64,9 @@ public RoutingService(Settings settings, ThreadPool threadPool, ClusterService c
         this.threadPool = threadPool;
         this.clusterService = clusterService;
         this.allocationService = allocationService;
-        this.schedule = settings.getAsTime("cluster.routing.schedule", timeValueSeconds(10));
-        clusterService.addFirst(this);
+        if (clusterService != null) {
+            clusterService.addFirst(this);
```

Comment:
at some point we should really add these ourside of the classes itself IMO. but that is totally unrelated